### PR TITLE
feat: json example

### PR DESCRIPTION
**Pull request by zhuliquan**
json parser example

### CODE REVIEW

File: _examples/json/main.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+	jsonLexer = lexer.MustSimple([]lexer.SimpleRule{
+		{Name: "Comment", Pattern: `\/\/[^\n]*`},
+		{Name: "String", Pattern: `"(\\"|[^"])*"`},
+		{Name: "Number", Pattern: `[-+]?(\d*\.)?\d+`},
+		{Name: "Punct", Pattern: `[-[!@#$%^&*()+_={}\|:;"'<,>.?/]|]`},
+		{Name: "Null", Pattern: "null"},
+		{Name: "True", Pattern: "true"},
+		{Name: "False", Pattern: "false"},
+		{Name: "EOL", Pattern: `[\n\r]+`},
+		{Name: "Whitespace", Pattern: `[ \t\n\r]+`},
```
Comment (alecthomas) [inline _examples/json/main.go:23]: `\n\r` will never match here because `EOL` takes precedence.

File: _examples/json/main.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+	"github.com/alecthomas/participle/v2"
+	"github.com/alecthomas/participle/v2/lexer"
+)
+
+var (
+	jsonLexer = lexer.MustSimple([]lexer.SimpleRule{
+		{Name: "Comment", Pattern: `\/\/[^\n]*`},
+		{Name: "String", Pattern: `"(\\"|[^"])*"`},
+		{Name: "Number", Pattern: `[-+]?(\d*\.)?\d+`},
```
Comment (alecthomas) [inline _examples/json/main.go:17]: This is not 100% accurate, eg. it won't match `1e10`, but it's okay for an example.

**Review by alecthomas — APPROVED** [review]
Awesome thanks

File: _examples/json/main.go (hunk 2 above)
Comment (zhuliquan) [reply _examples/json/main.go:17]: I am not sure whether json support scientific notation

File: _examples/json/main.go (hunk 1 above)
Comment (zhuliquan) [reply _examples/json/main.go]: fixed it

File: _examples/json/main.go (hunk 2 above)
Comment (alecthomas) [reply _examples/json/main.go:17]: It definitely does.

**Pull request merged**